### PR TITLE
Fix DevPortal scope mapping issue

### PR DESCRIPTION
**Pull request by npamudika**
Fixes https://github.com/wso2/product-apim/issues/10426

Porting fixes from APIM 3.2.0

### CODE REVIEW

**Review by malinthaprasan — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx [hunk 1]
```diff
@@ -177,8 +177,13 @@ class AuthManager {
         if (user) {
             const userScopes = user.scopes;
             const validScope = APIClient.getScopeForResource(resourcePath, resourceMethod);
-            return validScope.then((scope) => {
-                return userScopes.includes(scope);
+            let validity = []
+            return validScope.then((scopes) => {
+                scopes.map(scope => {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx:182]: Declaring `validity` array outside the `then` callback function scope is not valid, 

You can refer to [this implementation](https://github.com/wso2/carbon-apimgt/pull/10404/files#diff-108a02ceb37d6f854480c966d632c0f0f671dea4df9b417fca76271d2676d2ebR89-R95)

**Review by tmkasun — CHANGES_REQUESTED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx [hunk 2]
```diff
@@ -177,8 +177,13 @@ class AuthManager {
...
+                    const isValid = userScopes.includes(scope);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx:183]: This iterates through the entire scopes set even though we need to include only one of the defined scopes

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx [hunk 3]
```diff
@@ -177,8 +177,13 @@ class AuthManager {
...
+                    validity.push(isValid)
+                });
+                return validity.includes(false) ? false : true
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx:186]: This is AND logic right ? But scopes in REST API has OR meaning , I:e `apim:api_create,apim:api_delete` means at least one of the @malinthaprasan correct me if I'm wrong ?

Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx:186]: Hmm, yes. I think we should check for OR here. 
```
return validity.includes(true) ? true : false
```

**Comment by npamudika** [comment]
@kavishkafernando is working on the requested changes and will update the PR once that is done.

**Comment by tharikaGitHub** [comment]
This fix is added with this PR - https://github.com/wso2/carbon-apimgt/pull/10476. Hence closing.

**Pull request closed without merging**